### PR TITLE
support missing cases for sweep tests

### PR DESCRIPTION
**Pull request by asandhupatlaTT**
### Ticket
Link to Github Issue
https://github.com/tenstorrent/tt-metal/issues/15549

### Problem description
Provide context for the problem.
Support various  combinations of missing test cases

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.
Add all missing combinations of test case

### Checklist
- [x] Post commit CI passes : https://github.com/tenstorrent/tt-metal/actions/runs/12396806647
- [x] Blackhole Post commit (if applicable) : https://github.com/tenstorrent/tt-metal/actions/runs/12396812182
- [x] Model regression CI testing passes (if applicable) : https://github.com/tenstorrent/tt-metal/actions/runs/12396814398
- [x] Device performance regression CI testing passes (if applicable) : https://github.com/tenstorrent/tt-metal/actions/runs/12396816791
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/sweep_framework/sweeps/reduction/mean/mean.py [hunk 1]
```diff
@@ -27,15 +27,37 @@
 # Developers can create their own generator functions and pass them to the parameters as inputs.
 parameters = {
     "nightly": {
-        "input_shape": gen_shapes([1, 1, 1, 1], [2, 6, 128, 128], [1, 1, 32, 32], 32),
-        "input_a_dtype": [ttnn.bfloat16, ttnn.bfloat8_b],
-        "input_layout": [ttnn.TILE_LAYOUT],
+        "input_shape": gen_shapes([1, 1, 1, 1], [2, 6, 128, 128], [1, 1, 32, 32], 32)
+        + gen_shapes([1, 1, 1, 1], [2, 6, 110, 130], [1, 1, 32, 32], 31)
```
Comment (bbradelTT) [inline tests/sweep_framework/sweeps/reduction/mean/mean.py:31]: Are the same vectors generated each time or are they different?

File: tests/sweep_framework/sweeps/reduction/mean/mean.py [hunk 2]
```diff
@@ -27,15 +27,37 @@
...
+        + gen_shapes([1, 1, 1], [2, 6, 128], [1, 1, 32], 8)
+        + gen_shapes([1, 1, 1], [2, 7, 100], [1, 1, 64], 16)
+        + gen_shapes([1, 1, 1], [2, 8, 255], [1, 1, 122], 17)
+        + gen_shapes([1, 1], [2, 26], [1, 11], 2)
+        + gen_shapes([1, 1], [2, 36], [1, 12], 3)
+        + gen_shapes([1, 1], [2, 46], [1, 13], 4)
+        + gen_shapes([1], [2, 6], [1, 1], 2)
+        + gen_shapes([1], [2, 6], [1, 1], 2)
+        + gen_shapes([1], [10, 10], [2, 3], 2),
+        "input_a_dtype": [ttnn.float32, ttnn.bfloat16, ttnn.bfloat8_b],
```
Comment (bbradelTT) [inline tests/sweep_framework/sweeps/reduction/mean/mean.py:42]: in the run command you'll need to check for float32 at the start and not execute the rest of the code for GS

File: tests/sweep_framework/sweeps/reduction/mean/mean.py [hunk 3]
```diff
@@ -27,15 +27,37 @@
...
+        "input_layout": [ttnn.ROW_MAJOR_LAYOUT, ttnn.TILE_LAYOUT],
```
Comment (bbradelTT) [inline tests/sweep_framework/sweeps/reduction/mean/mean.py:43]: Also need to add dim, and keepdim as parameters

And throughout dim needs to be multiple combinations as well

File: tests/sweep_framework/sweeps/reduction/mean/mean.py (hunk 1 above)
Comment (asandhupatlaTT) [reply tests/sweep_framework/sweeps/reduction/mean/mean.py:31]: they are always same since we fix the seed with deterministic value

File: tests/sweep_framework/sweeps/reduction/traces/max_traces.py [hunk 4]
```diff
@@ -15,24 +15,46 @@
 
 parameters = {
     "default": {
-        "height": [25],
-        "width": [4],
-        "dtype": [ttnn.float32, ttnn.bfloat16],
+        "height": [25, 16],
```
Comment (bbradelTT) [inline tests/sweep_framework/sweeps/reduction/traces/max_traces.py:18]: Please revert the changes for the traces files (argmax_traces.py and max_traces.py). These are from traces, and should only change when traces change or new traces are added.

**Review by bbradelTT — APPROVED** [review]

**Review by jdesousa-TT — APPROVED** [review]
Please test these using the sweeps pipeline
